ls1intum/Artemis
# `Assessment`: Fix the availability of the filter for locked submissions

### Checklist
#### General


#### Server
#### Client


### Motivation and Context


The issue [#5307](https://github.com/ls1intum/Artemis/issues/5307) presents the previous behavior. 
### Description


The filter for locked submissions is available just if there are already locked submissions. Otherwise no. 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Modeling/text/programming/upload file exercise 

1. Log in to Artemis (as instructor/tutor)
2. Navigate to Course Administration
3. Create 1 Modeling, 1 text Exercise, 1 programming exercise, and 1 upload file exercise ( all of them with the manual assessment).
{First case}
4. Check that without submissions, the filter for locked submissions **is not available** (_see screenshot 1_). 
{Second case}
4. As a student submit the solution for those exercises 
6. As an instructor, do not provide any assessment yet and don't lock any submission, and check the filter for locked submissions **is not available** (_see screenshot 2_).
{Third case}
7. Open some submissions and locked them. Then verify that the filter for locked submissions **is available** now (_see screenshot 3_). 
{Fourth case}
8. Unlock or complete the assessment for those submissions and check the filter for locked submissions **is not available** (_see screenshot 4_).


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots


Screenshot 1
<img width="831" alt="image" src="https://user-images.githubusercontent.com/88991954/176648207-fbdb4fda-8c6f-4349-b990-aeb7a0ec4988.png">

Screenshot 2
<img width="1500" alt="image" src="https://user-images.githubusercontent.com/88991954/176659409-1a29ed43-4867-49bf-8d54-08fde3fe560d.png">

Screenshot 3
<img width="1504" alt="image" src="https://user-images.githubusercontent.com/88991954/176649360-922df9a3-8cf7-4870-9ac5-aba09347e5af.png">


Screenshot 4
<img width="1490" alt="image" src="https://user-images.githubusercontent.com/88991954/176649083-fd7efbb3-fc82-47b0-9d1d-5cbed8dc0391.png">
